pkgs/applications/networking/instant-messengers/qq/update.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p curl nix-prefetch-git jq
+
+set -euo pipefail
+
+cd $(readlink -e $(dirname "${BASH_SOURCE[0]}"))
+
+nixpkgs="$(git rev-parse --show-toplevel || (printf 'Could not find root of nixpkgs repo\nAre we running from within the nixpkgs git repo?\n' >&2; exit 1))"
+
+nixeval() {
+    nix --extra-experimental-features nix-command eval --json --impure -f "$nixpkgs" "$1" | jq -r .
+}
+
+payload=$(curl https://im.qq.com/rainbow/linuxQQDownload | grep -oP "var params= \K\{.*\}(?=;)")
+amd64_url=$(jq -r .x64DownloadUrl.deb <<< "$payload")
+arm64_url=$(jq -r .armDownloadUrl.deb <<< "$payload")
+
+urlhash=$(grep -oP "(?<=QQNT/)[a-e0-9]+(?=/linuxqq)" <<< "$amd64_url")
+version=$(grep -oP "(?<=/linuxqq_).*(?=_amd64.deb)" <<< "$amd64_url")
+
+attr="${UPDATE_NIX_ATTR_PATH:-qq}"
+
+curversion="$(nixeval "$attr.version")"
+
+if [[ "$curversion" == "$version" ]]; then
+  echo "Already up to date!"
+  exit 0
+fi
+
+sed -i -E -e "s#version = \".*\"#version = \"$version\"#" default.nix
+sed -i -E -e "s#_hash = \".*\"#_hash = \"$urlhash\"#" default.nix
+
+amd64_hash=$(nix-prefetch-url $amd64_url)
+arm64_hash=$(nix-prefetch-url $arm64_url)
+
+# use friendlier hashes
+amd64_hash=$(nix hash to-sri --type sha256 "$amd64_hash")
+arm64_hash=$(nix hash to-sri --type sha256 "$arm64_hash")
+
+sed -i -E -e "/_amd64.deb\";$/{N
+s#hash = \".*\"#hash = \"$amd64_hash\"#
+}" default.nix
+
+sed -i -E -e "/_arm64.deb\";$/{N
+s#hash = \".*\"#hash = \"$arm64_hash\"#
+}" default.nix
```

Comment:
While well written, i'd say it be much more robust to dump these values into a json or separate nix file which is then loaded/imported into the derivation

Reply:
or, consider `common-updater-scripts`